cpp/src/fil/infer.cu

```diff
@@ -447,11 +450,13 @@ struct tree_aggregator_t<NITEMS, CATEGORICAL_LEAF> {
     // __syncthreads() is called in infer_k
   }
   __device__ __forceinline__ void accumulate(
-    vec<NITEMS, int> single_tree_prediction, int tree) {
+    vec<NITEMS, int> single_tree_prediction, int tree, int num_rows) {
 #pragma unroll
-    for (int item = 0; item < NITEMS; ++item)
-      raft::myAtomicAdd(votes + single_tree_prediction[item] * NITEMS + item,
-                        1);
+    for (int item = 0; item < NITEMS; ++item) {
+      if (item < num_rows)
```

Comment:
fixing a bug which somehow did not trigger the test in the previous PR. It triggers it now.